subsys/pcd/src/pcd.c

```diff
@@ -174,10 +166,27 @@ static int network_core_update(const void *src_addr, size_t len, bool wait)
 
```

Comment:
If (!wait) is true, then I assume this is supposed to be non-blocking.

In which case we will now be removing the rights of the network core while it is asynchronously doing it's job.

I think this will fail.

Reply:
If it's not supposed to wait it will exit which is the intended functionality. The actual asynchronous wait is done in the `do while` loop which is just using `k_busy_wait` suspending the thread in the function meaning that it won't exit. I don't think it will fail.

Reply:
Here is the (new with this patch) scenario I suspect will fail:

elavate rights of network core
start work on network core
return due to (!wait)
reduce rights of network core
network core accesses the secure pcd_sram as non-secure and fails

Reply:
From what I can see there is 2 different version of the function 
network_core_update_initiate
and
network_core_update

The only use I've found is https://github.com/nrfconnect/sdk-nrf/blob/3b5a72f1eb0714610feee46a416a453218841f5d/modules/mcuboot/hooks/nrf53_hooks.c#L97-L104

I agree with you that changing the premisions of the network core while it's running could be a bad thing. Didn't see that `!wait` allows the network core to be turned on but never off again.

Reply:
I can't find a place where the `!wait` is used https://github.com/nrfconnect/sdk-nrf/blob/3b5a72f1eb0714610feee46a416a453218841f5d/modules/mcuboot/hooks/nrf53_hooks.c#L114

Reply:
@hakonfam I think we could remove this functionality, I don't really see the purpose of it when it's never used.